test/e2e-go/restAPI/restClient_test.go

```diff
@@ -1227,3 +1226,264 @@ func TestNilStateProofInParticipationInfo(t *testing.T) {
+		wh, nil, someAddress, createdAppID.Address().String(),
+		0, 10_000_000, nil, "", 0, 0,
+	)
+	a.NoError(err)
+	appFundTxID := appFundTxn.ID()
+	_, err = waitForTransaction(t, testClient, someAddress, appFundTxID.String(), 30*time.Second)
+	a.NoError(err)
+
+	createdBoxName := map[string]bool{}
+	var createdBoxCount uint64 = 0
+
+	// define operate box helper
+	operateBoxAndSendTxn := func(operation string, boxNames []string) {
+		txns := make([]transactions.Transaction, len(boxNames))
+		txIDs := make(map[string]string, len(boxNames))
+
+		for i := 0; i < len(boxNames); i++ {
+			appArgs := [][]byte{
+				[]byte(operation),
+				[]byte(boxNames[i]),
+			}
+			boxRef := transactions.BoxRef{
+				Name:  []byte(boxNames[i]),
+				Index: 0,
+			}
+
+			txns[i], err = testClient.MakeUnsignedAppNoOpTx(
+				uint64(createdAppID), appArgs,
+				nil, nil, nil,
+				[]transactions.BoxRef{boxRef},
+			)
+			a.NoError(err)
+			txns[i], err = testClient.FillUnsignedTxTemplate(someAddress, 0, 0, 0, txns[i])
+			a.NoError(err)
+			txIDs[txns[i].ID().String()] = someAddress
+		}
+
+		var gid crypto.Digest
+		gid, err = testClient.GroupID(txns)
+		a.NoError(err)
+
+		stxns := make([]transactions.SignedTxn, len(boxNames))
+		for i := 0; i < len(boxNames); i++ {
+			txns[i].Group = gid
+			wh, err = testClient.GetUnencryptedWalletHandle()
+			a.NoError(err)
+			stxns[i], err = testClient.SignTransactionWithWallet(wh, nil, txns[i])
+			a.NoError(err)
+		}
+
+		err = testClient.BroadcastTransactionGroup(stxns)
+		a.NoError(err)
+
+		_, err = waitForTransaction(t, testClient, someAddress, txns[0].ID().String(), 30*time.Second)
+		a.NoError(err)
+		return
+	}
+
+	operateAndMatchRes := func(operation string, boxNames ...string) {
+		if operation == "create" {
+			for _, box := range boxNames {
+				keyValid, ok := createdBoxName[box]
+				a.False(ok && keyValid)
+			}
+		} else if operation == "delete" {
+			for _, box := range boxNames {
+				keyValid, ok := createdBoxName[box]
+				a.True(keyValid == ok)
+			}
+		} else {
+			a.True(false)
+		}
+
+		operateBoxAndSendTxn(operation, boxNames)
+
+		if operation == "create" {
+			for _, box := range boxNames {
+				createdBoxName[box] = true
+			}
+			createdBoxCount += uint64(len(boxNames))
+		} else if operation == "delete" {
+			for _, box := range boxNames {
+				createdBoxName[box] = false
+			}
+			createdBoxCount -= uint64(len(boxNames))
+		}
+
+		resp, err := testClient.ApplicationBoxes(uint64(createdAppID))
+		a.NoError(err)
+		a.Equal(createdBoxCount, uint64(len(resp)))
+		for _, byteName := range resp {
+			a.True(createdBoxName[string(byteName)])
+		}
+	}
+
+	resp, err := testClient.ApplicationBoxes(uint64(createdAppID))
+	a.NoError(err)
+	a.Empty(resp)
+
+	operateAndMatchRes("create",
+		` `,
+		`     	`,
+		` ? = % ;`,
+		`; DROP *;`,
+		`OR 1 = 1;`,
+		`"      ;  SELECT * FROM kvstore; DROP acctrounds; `,
+		`背负青天而莫之夭阏者，而后乃今将图南。`,
+	)
+	operateAndMatchRes("create",
+		`那些智力超常的人啊`,
```

Comment:
nit: I like the db commands - I think it would be nice if there were a `const` or in a `[]string` and loop through the tests, then no need to repeat the literal in the `delete`. 

e.g. r
```
tests := []string{{"abc", ...}}
for _, test := range tests {
   operateAndMatchRes(test)
   //check or do something, etc.
}
```
Also suggest something with non UTF-8 like null chars or `0xF8~0xFF`, e.g. `"create", []byte([0, 255, 254, 254])`

Reply:
yea, using `const` or sharing slice of `string` would be making sense, the UTF-8 suggestion is also making sense, that can also live in the constant slice of `string`.

Reply:
sorry, I think the suggestion should be to cast the byte array to `string` (I think boxes take in `string` - although I think they are roughly equivalent?).

Reply:
I got it, it just wrapping around byte array/slice with `string`, but they should be just roughly similar (I agree).

Reply:
Thanks for the tests, looks good to resolve